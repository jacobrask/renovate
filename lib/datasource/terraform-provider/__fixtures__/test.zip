version https://git-lfs.github.com/spec/v1
oid sha256:b7871907b3f8222cda1cfb48930d96e89d82adf965da3e934036488be64a475b
size 618
