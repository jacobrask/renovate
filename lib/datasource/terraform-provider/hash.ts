import crypto from 'crypto';
import * as fs from 'fs';
import extract from 'extract-zip';
import { logger } from '../../logger';

export function hashFiles(files: string[]): string {
  const rootHash = crypto.createHash('sha256');

  files.forEach((file) => {
    // build for every file a line looking like "aaaaaaaaaaaaaaa  file.txt\n"
    const hash = crypto.createHash('sha256');

    // a sha256sum displayed as lowercase hex string to root hash
    const fileBuffer = fs.readFileSync(file);
    hash.update(fileBuffer);
    hash.end();
    const data = hash.read();
    rootHash.update(data.toString('hex'));

    // add double space, the filename and a new line char
    rootHash.update('  ');
    const fileName = file.replace(/^.*[\\/]/, '');
    rootHash.update(fileName);
    rootHash.update('\n');
  });

  rootHash.end();
  const rootData = rootHash.read();
  const result: string = rootData.toString('base64');
  return `h1:${result}`;
}

export async function hashOfZipContent(
  zipFilePath: string,
  extractPath: string
): Promise<string> {
  await extract(zipFilePath, { dir: extractPath });
  const files = fs.readdirSync(extractPath);
  // the h1 hashing algorithms requires that the files are sorted by filename
  const sortedFiles = files.sort((a, b) => a.localeCompare(b));
  const filesWithPath = sortedFiles.map((file) => `${extractPath}/${file}`);

  const result = hashFiles(filesWithPath);

  // delete extracted files
  filesWithPath.forEach((value) =>
    fs.unlink(value, (err) =>
      logger.warn({ err }, 'Failed to delete extracted file')
    )
  );
  return result;
}
