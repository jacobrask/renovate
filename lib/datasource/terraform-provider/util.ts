export const RepositoryRegex = /^(?:(?<namespace>[^\s]+)\/)?(?<dependency>[^\s]+)$/;
