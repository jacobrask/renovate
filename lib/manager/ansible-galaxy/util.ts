export const newBlockRegEx = /^\s*-\s*((\w+):\s*(.*))$/;
export const blockLineRegEx = /^\s*((\w+):\s*(.*))$/;
export const galaxyDepRegex = /.+\..+/;
