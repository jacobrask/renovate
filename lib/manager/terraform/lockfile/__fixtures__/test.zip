version https://git-lfs.github.com/spec/v1
oid sha256:1721d74eaf36c245148394532dc93854e5568352efad27e57361864e7edfded7
size 473638
