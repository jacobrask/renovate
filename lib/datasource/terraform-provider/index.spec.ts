import fs from 'fs';
import { getPkgReleases } from '..';
import * as httpMock from '../../../test/http-mock';
import { getName } from '../../../test/util';
import { hashOfZipContent } from './hash';
import { createReleases, id as datasource, defaultRegistryUrls } from '.';

const consulData: any = fs.readFileSync(
  'lib/datasource/terraform-provider/__fixtures__/azurerm-provider.json'
);
const hashicorpReleases: any = fs.readFileSync(
  'lib/datasource/terraform-provider/__fixtures__/releaseBackendIndex.json'
);
const serviceDiscoveryResult: any = fs.readFileSync(
  'lib/datasource/terraform-module/__fixtures__/service-discovery.json'
);

const releaseBackend: any = fs.readFileSync(
  'lib/datasource/terraform-provider/__fixtures__/releaseBackendFormatted.json'
);

const randomLinuxAmd64: Buffer = fs.readFileSync(
  'lib/datasource/terraform-provider/__fixtures__/test.zip'
);

const primaryUrl = defaultRegistryUrls[0];
const secondaryUrl = defaultRegistryUrls[1];

describe(getName(__filename), () => {
  jest.setTimeout(15000);

  describe('createReleases', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      httpMock.setup();
    });

    afterEach(() => {
      httpMock.reset();
    });
    it('testing', async () => {
      httpMock
        .scope(secondaryUrl)
        .get('/index.json')
        .reply(200, releaseBackend)
        .get(
          '/terraform-provider-azurerm/2.53.0/terraform-provider-azurerm_2.53.0_linux_amd64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-azurerm/2.53.0/terraform-provider-azurerm_2.53.0_linux_arm64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-azurerm/2.53.0/terraform-provider-azurerm_2.53.0_windows_amd64.zip'
        )
        .reply(200, randomLinuxAmd64);

      const result = await createReleases(
        'azurerm',
        defaultRegistryUrls[1],
        'hashicorp/azurerm',
        ['2.53.0']
      );
      expect(result).not.toBeNull();
      expect(result).toBeArrayOfSize(1);
      expect(result[0].hashes).toBeArrayOfSize(3);
      expect(result[0].hashes).toEqual([
        { linux_amd64: 'h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=' },
        { linux_arm64: 'h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=' },
        { windows_amd64: 'h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=' },
      ]);
    });
  });
  describe('hashOfZipContent()', () => {
    it('full data', async () => {
      const result = await hashOfZipContent(
        'lib/datasource/terraform-provider/__fixtures__/test.zip',
        '/tmp/test'
      );
      expect(result).not.toBeNull();
      expect(result).toEqual('h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=');
    });
  });
  describe('getReleases', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      httpMock.setup();
    });

    afterEach(() => {
      httpMock.reset();
    });

    it('returns null for empty result', async () => {
      httpMock
        .scope(primaryUrl)
        .get('/v1/providers/hashicorp/azurerm')
        .reply(200, {})
        .get('/.well-known/terraform.json')
        .reply(200, serviceDiscoveryResult);
      httpMock.scope(secondaryUrl).get('/index.json').reply(200, {});
      expect(
        await getPkgReleases({
          datasource,
          depName: 'azurerm',
        })
      ).toBeNull();
      expect(httpMock.getTrace()).toMatchSnapshot();
    });
    it('returns null for 404', async () => {
      httpMock
        .scope(primaryUrl)
        .get('/v1/providers/hashicorp/azurerm')
        .reply(404)
        .get('/.well-known/terraform.json')
        .reply(200, serviceDiscoveryResult);
      httpMock.scope(secondaryUrl).get('/index.json').reply(404);
      expect(
        await getPkgReleases({
          datasource,
          depName: 'azurerm',
        })
      ).toBeNull();
      expect(httpMock.getTrace()).toMatchSnapshot();
    });
    it('returns null for unknown error', async () => {
      httpMock
        .scope(primaryUrl)
        .get('/v1/providers/hashicorp/azurerm')
        .replyWithError('')
        .get('/.well-known/terraform.json')
        .reply(200, serviceDiscoveryResult);
      httpMock.scope(secondaryUrl).get('/index.json').replyWithError('');
      expect(
        await getPkgReleases({
          datasource,
          depName: 'azurerm',
        })
      ).toBeNull();
      expect(httpMock.getTrace()).toMatchSnapshot();
    });
    it('processes real data', async () => {
      httpMock
        .scope(primaryUrl)
        .get('/v1/providers/hashicorp/azurerm')
        .reply(200, JSON.parse(consulData))
        .get('/.well-known/terraform.json')
        .reply(200, serviceDiscoveryResult);
      httpMock
        .scope(secondaryUrl)
        .get('/index.json')
        .reply(200, releaseBackend)
        .get(
          '/terraform-provider-azurerm/2.53.0/terraform-provider-azurerm_2.53.0_linux_amd64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-azurerm/2.53.0/terraform-provider-azurerm_2.53.0_linux_arm64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-azurerm/2.53.0/terraform-provider-azurerm_2.53.0_windows_amd64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-azurerm/2.52.0/terraform-provider-azurerm_2.52.0_linux_amd64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-azurerm/2.52.0/terraform-provider-azurerm_2.52.0_linux_arm64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-azurerm/2.52.0/terraform-provider-azurerm_2.52.0_windows_amd64.zip'
        )
        .reply(200, randomLinuxAmd64);
      const res = await getPkgReleases({
        datasource,
        depName: 'azurerm',
      });
      expect(res).toMatchSnapshot();
      expect(res).not.toBeNull();
    });

    it('processes real data from lookupName, only return not all versions', async () => {
      httpMock
        .scope('https://registry.company.com')
        .get('/v1/providers/hashicorp/azurerm')
        .reply(200, JSON.parse(consulData))
        .get('/.well-known/terraform.json')
        .reply(200, serviceDiscoveryResult);
      httpMock
        .scope(secondaryUrl)
        .get('/index.json')
        .reply(200, releaseBackend)
        .get(
          '/terraform-provider-azurerm/2.53.0/terraform-provider-azurerm_2.53.0_linux_amd64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-azurerm/2.53.0/terraform-provider-azurerm_2.53.0_linux_arm64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-azurerm/2.53.0/terraform-provider-azurerm_2.53.0_windows_amd64.zip'
        )
        .reply(200, randomLinuxAmd64);

      const res = await getPkgReleases({
        datasource,
        depName: 'azure',
        lookupName: 'hashicorp/azurerm',
        registryUrls: ['https://registry.company.com'],
      });
      expect(res).toMatchSnapshot();
      expect(res).not.toBeNull();
    });
    it('processes data with alternative backend', async () => {
      httpMock
        .scope(primaryUrl)
        .get('/v1/providers/hashicorp/google-beta')
        .reply(404, {
          errors: ['Not Found'],
        })
        .get('/.well-known/terraform.json')
        .reply(200, serviceDiscoveryResult);
      httpMock
        .scope(secondaryUrl)
        .get('/index.json')
        .reply(200, JSON.parse(hashicorpReleases))
        .get(
          '/terraform-provider-google-beta/1.19.0/terraform-provider-google-beta_1.19.0_darwin_amd64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-google-beta/1.19.0/terraform-provider-google-beta_1.19.0_freebsd_386.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-google-beta/1.19.0/terraform-provider-google-beta_1.19.0_freebsd_amd64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-google-beta/1.20.0/terraform-provider-google-beta_1.20.0_darwin_amd64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-google-beta/1.20.0/terraform-provider-google-beta_1.20.0_freebsd_386.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-google-beta/1.20.0/terraform-provider-google-beta_1.20.0_freebsd_amd64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-google-beta/2.0.0/terraform-provider-google-beta_2.0.0_darwin_amd64.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-google-beta/2.0.0/terraform-provider-google-beta_2.0.0_freebsd_386.zip'
        )
        .reply(200, randomLinuxAmd64)
        .get(
          '/terraform-provider-google-beta/2.0.0/terraform-provider-google-beta_2.0.0_freebsd_amd64.zip'
        )
        .reply(200, randomLinuxAmd64);

      const res = await getPkgReleases({
        datasource,
        depName: 'google-beta',
      });
      expect(res).toMatchSnapshot();
      expect(res).not.toBeNull();
    });
    it('simulate failing secondary release source', async () => {
      httpMock
        .scope(primaryUrl)
        .get('/v1/providers/hashicorp/google-beta')
        .reply(404, {
          errors: ['Not Found'],
        })
        .get('/.well-known/terraform.json')
        .reply(200, serviceDiscoveryResult);
      httpMock.scope(secondaryUrl).get('/index.json').reply(404);

      const res = await getPkgReleases({
        datasource,
        depName: 'datadog',
      });
      expect(res).toMatchSnapshot();
      expect(res).toBeNull();
    });
    it('returns null for error in service discovery', async () => {
      httpMock.scope(primaryUrl).get('/.well-known/terraform.json').reply(404);
      httpMock.scope(secondaryUrl).get('/index.json').replyWithError('');
      expect(
        await getPkgReleases({
          datasource,
          depName: 'azurerm',
        })
      ).toBeNull();
      expect(httpMock.getTrace()).toMatchSnapshot();
    });
  });
});
