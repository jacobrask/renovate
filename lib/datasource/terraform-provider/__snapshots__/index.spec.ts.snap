// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`datasource/terraform-provider/index getReleases processes data with alternative backend 1`] = `
Object {
  "registryUrl": "https://releases.hashicorp.com",
  "releases": Array [
    Object {
      "version": "1.19.0",
    },
    Object {
      "version": "1.20.0",
    },
    Object {
      "version": "2.0.0",
    },
  ],
  "sourceUrl": "https://github.com/terraform-providers/terraform-provider-google-beta",
}
`;

exports[`datasource/terraform-provider/index getReleases processes data with alternative backend 2`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/v1/providers/hashicorp/google-beta",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "releases.hashicorp.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://releases.hashicorp.com/index.json",
  },
]
`;

exports[`datasource/terraform-provider/index getReleases processes real data 1`] = `
Object {
  "homepage": "https://registry.terraform.io/providers/hashicorp/azurerm",
  "registryUrl": "https://registry.terraform.io",
  "releases": Array [
    Object {
      "version": "2.52.0",
    },
    Object {
      "releaseTimestamp": "2019-11-26T08:22:56.000Z",
      "version": "2.53.0",
    },
  ],
  "sourceUrl": "https://github.com/terraform-providers/terraform-provider-azurerm",
}
`;

exports[`datasource/terraform-provider/index getReleases processes real data 2`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/v1/providers/hashicorp/azurerm",
  },
]
`;

exports[`datasource/terraform-provider/index getReleases processes real data from lookupName 1`] = `
Object {
  "homepage": "https://registry.company.com/providers/hashicorp/azurerm",
  "registryUrl": "https://registry.company.com",
  "releases": Array [
    Object {
      "version": "2.52.0",
    },
    Object {
      "releaseTimestamp": "2019-11-26T08:22:56.000Z",
      "version": "2.53.0",
    },
  ],
  "sourceUrl": "https://github.com/terraform-providers/terraform-provider-azurerm",
}
`;

exports[`datasource/terraform-provider/index getReleases processes real data from lookupName 2`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.company.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.company.com/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.company.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.company.com/v1/providers/hashicorp/azurerm",
  },
]
`;

exports[`datasource/terraform-provider/index getReleases returns null for 404 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/v1/providers/hashicorp/azurerm",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "releases.hashicorp.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://releases.hashicorp.com/index.json",
  },
]
`;

exports[`datasource/terraform-provider/index getReleases returns null for empty result 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/v1/providers/hashicorp/azurerm",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "releases.hashicorp.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://releases.hashicorp.com/index.json",
  },
]
`;

exports[`datasource/terraform-provider/index getReleases returns null for error in service discovery 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "releases.hashicorp.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://releases.hashicorp.com/index.json",
  },
]
`;

exports[`datasource/terraform-provider/index getReleases returns null for unknown error 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/v1/providers/hashicorp/azurerm",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "releases.hashicorp.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://releases.hashicorp.com/index.json",
  },
]
`;

exports[`datasource/terraform-provider/index getReleases simulate failing secondary release source 1`] = `null`;

exports[`test getReleases processes data with alternative backend 1`] = `
Object {
  "registryUrl": "https://releases.hashicorp.com",
  "releases": Array [
    Object {
      "hashes": Object {
        "darwin_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "freebsd_386": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "freebsd_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
      },
      "version": "1.19.0",
    },
    Object {
      "hashes": Object {
        "darwin_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "freebsd_386": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "freebsd_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
      },
      "version": "1.20.0",
    },
    Object {
      "hashes": Object {
        "darwin_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "freebsd_386": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "freebsd_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
      },
      "version": "2.0.0",
    },
  ],
  "sourceUrl": "https://github.com/terraform-providers/terraform-provider-google-beta",
}
`;

exports[`test getReleases processes real data 1`] = `
Object {
  "homepage": "https://registry.terraform.io/providers/hashicorp/azurerm",
  "registryUrl": "https://registry.terraform.io",
  "releases": Array [
    Object {
      "hashes": Object {
        "linux_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "linux_arm64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "windows_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
      },
      "version": "2.52.0",
    },
    Object {
      "hashes": Object {
        "linux_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "linux_arm64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "windows_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
      },
      "releaseTimestamp": "2019-11-26T08:22:56.000Z",
      "version": "2.53.0",
    },
  ],
  "sourceUrl": "https://github.com/terraform-providers/terraform-provider-azurerm",
}
`;

exports[`test getReleases processes real data from lookupName, only return not all versions 1`] = `
Object {
  "homepage": "https://registry.company.com/providers/hashicorp/azurerm",
  "registryUrl": "https://registry.company.com",
  "releases": Array [
    Object {
      "hashes": Object {},
      "version": "2.52.0",
    },
    Object {
      "hashes": Object {
        "linux_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "linux_arm64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "windows_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
      },
      "releaseTimestamp": "2019-11-26T08:22:56.000Z",
      "version": "2.53.0",
    },
  ],
  "sourceUrl": "https://github.com/terraform-providers/terraform-provider-azurerm",
}
`;

exports[`test getReleases processes real data from lookupName, return not all versions 1`] = `
Object {
  "homepage": "https://registry.company.com/providers/hashicorp/azurerm",
  "registryUrl": "https://registry.company.com",
  "releases": Array [
    Object {
      "hashes": Object {},
      "version": "2.52.0",
    },
    Object {
      "hashes": Object {
        "linux_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "linux_arm64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
        "windows_amd64": "h1:CgObCCvnD/qDmF6aOYQ2KvB2arDbcBURxARZfYB9Mvc=",
      },
      "releaseTimestamp": "2019-11-26T08:22:56.000Z",
      "version": "2.53.0",
    },
  ],
  "sourceUrl": "https://github.com/terraform-providers/terraform-provider-azurerm",
}
`;

exports[`test getReleases returns null for 404 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/v1/providers/hashicorp/azurerm",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "releases.hashicorp.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://releases.hashicorp.com/index.json",
  },
]
`;

exports[`test getReleases returns null for empty result 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/v1/providers/hashicorp/azurerm",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "releases.hashicorp.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://releases.hashicorp.com/index.json",
  },
]
`;

exports[`test getReleases returns null for error in service discovery 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "releases.hashicorp.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://releases.hashicorp.com/index.json",
  },
]
`;

exports[`test getReleases returns null for unknown error 1`] = `
Array [
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/.well-known/terraform.json",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "registry.terraform.io",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://registry.terraform.io/v1/providers/hashicorp/azurerm",
  },
  Object {
    "headers": Object {
      "accept": "application/json",
      "accept-encoding": "gzip, deflate",
      "host": "releases.hashicorp.com",
      "user-agent": "https://github.com/renovatebot/renovate",
    },
    "method": "GET",
    "url": "https://releases.hashicorp.com/index.json",
  },
]
`;

exports[`test getReleases simulate failing secondary release source 1`] = `null`;
